enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/openshift/DeploymentConfigEnricher.java

```diff
@@ -0,0 +1,191 @@
+                }
+            }
+            DeploymentStrategy strategy = spec.getStrategy();
+            String strategyType = null;
+            if (strategy != null) {
+                strategyType = strategy.getType();
+            }
+            if (openshiftDeployTimeoutSeconds != null && openshiftDeployTimeoutSeconds > 0) {
+                if (StringUtils.isBlank(strategyType) || "Rolling".equals(strategyType)) {
+                    specBuilder.withNewStrategy().withType("Rolling").
+                            withNewRollingParams().withTimeoutSeconds(openshiftDeployTimeoutSeconds).endRollingParams().endStrategy();
+                } else if ("Recreate".equals(strategyType)) {
+                    specBuilder.withNewStrategy().withType("Recreate").
+                            withNewRecreateParams().withTimeoutSeconds(openshiftDeployTimeoutSeconds).endRecreateParams().endStrategy();
+                } else {
+                    specBuilder.withNewStrategy().withType(strategyType).endStrategy();
+                }
+            } else if (StringUtils.isNotBlank(strategyType)) {
+                // TODO is there any values we can copy across?
+                specBuilder.withNewStrategy().withType(strategyType).endStrategy();
+            }
+
+            if(enableAutomaticTrigger) {
+                specBuilder.addNewTrigger().withType("ConfigChange").endTrigger();
+            }
+
+            // add a new image change trigger for the build stream
+            if (containerToImageMap.size() != 0) {
+                if(runtimeMode.equals(RuntimeMode.openshift)) {
+                    for (Map.Entry<String, String> entry : containerToImageMap.entrySet()) {
+                        String containerName = entry.getKey();
+                        ImageName image = new ImageName(entry.getValue());
+                        String tag = image.getTag() != null ? image.getTag() : "latest";
+                        specBuilder.addNewTrigger()
+                                .withType("ImageChange")
+                                .withNewImageChangeParams()
+                                .withAutomatic(true)
+                                .withNewFrom()
+                                .withKind("ImageStreamTag")
+                                .withName(image.getSimpleName() + ":" + tag)
+                                .withNamespace(image.getUser())
+                                .endFrom()
+                                .withContainerNames(containerName)
+                                .endImageChangeParams()
+                                .endTrigger();
+                    }
+                }
+            }
+
+            specBuilder.endSpec();
+        }
+        return builder.build();
+    }
+
+    private Boolean isAutomaticTriggerEnabled(MavenEnricherContext enricherContext, Boolean defaultValue) {
+        if(enricherContext.getProperty("fabric8.openshift.enableAutomaticTrigger") != null) {
```

Comment:
If `getProperty` can be called without casting then we will not have a Maven dependency anymore.